docs/useCases.md

```diff
@@ -0,0 +1,632 @@
+- **limit**: (number) Limit for pagination.
+- **offset**: (number) Offset for pagination.
+- **fileSearchCriteria**: ([FileSearchCriteria](../src/files/domain/models/FileCriteria.ts)) Supports filtering the files by different file properties.
+- **fileOrderCriteria**: ([FileOrderCriteria](../src/files/domain/models/FileCriteria.ts)) Supports ordering the results according to different criteria. If not set, the defalt value is `FileOrderCriteria.NAME_AZ`.
+
+##### Example call using optional parameters:
+
+```typescript
+import { getDatasetFiles } from '@iqss/dataverse-client-javascript';
+
+/* ... */
+
+const datasetId: number = 2;
+const datasetVersionId: string = '1.0';
+const includeDeaccessioned: boolean = true;
+const limit: number = 10;
+const offset: number = 20;
+const searchCriteria: FileSearchCriteria = {
+  searchText: 'file title',
+};
+const orderCriteria: FileOrderCriteria = FileOrderCriteria.NEWEST;
+
+getDatasetFiles
+  .execute(datasetId, datasetVersionId, includeDeaccessioned, limit, offset, searchCriteria, orderCriteria)
+  .then((subset: FilesSubset) => {
+    /* ... */
+  });
+
+/* ... */
+```
+
+## Metadata Blocks
+
+### Metadata Blocks read use cases
+
+#### Get Metadata Block By Name
+
+Returns a [MetadataBlock](../src/metadataBlocks/domain/models/MetadataBlock.ts) instance, given its name.
+
+##### Example call:
+
+```typescript
+import { getMetadataBlockByName } from '@iqss/dataverse-client-javascript';
+
+/* ... */
+
+const name = 'citation';
+
+getMetadataBlockByName.execute(name).then((metadataBlock: MetadataBlock) => {
+  /* ... */
+});
+
+/* ... */
+```
+
+_See [use case](../src/metadataBlocks/domain/useCases/GetMetadataBlockByName.ts) implementation_.
+
+## Users
+
+### Users read use cases
+
+#### Get Current Authenticated User
+
+Returns the current [AuthenticatedUser](../src/users/domain/models/AuthenticatedUser.ts) corresponding to the authentication mechanism provided through `ApiConfig`.
+
+##### Example call:
+
+```typescript
+import { getCurrentAuthenticatedUser } from '@iqss/dataverse-client-javascript';
+
+/* ... */
+
+getCurrentAuthenticatedUser.execute().then((user: AuthenticatedUser) => {
+  /* ... */
+});
+
+/* ... */
+```
+
+_See [use case](../src/users/domain/useCases/GetCurrentAuthenticatedUser.ts) implementation_.
+
+## Info
+
+#### Get Dataverse Backend Version
+
+Returns a [DataverseVersion](../src/info/domain/models/DataverseVersion.ts) object, which contains version information for the Dataverse backend installation.
+
+##### Example call:
+
+```typescript
+import { getDataverseVersion } from '@iqss/dataverse-client-javascript';
+
+/* ... */
+
+getDataverseVersion.execute().then((version: DataverseVersion) => {
+  /* ... */
+});
+
+/* ... */
+```
+
+_See [use case](../src/info/domain/useCases/GetDataverseVersion.ts) implementation_.
+
+#### Get Maximum Embargo Duration In Months
+
+Returns a number indicating the configured maximum embargo duration in months.
+
```

Comment:
Does it make sense to add details about the possible return values? For example, in the Dataverse configuration docs, it says:
 

> A value of 0 months or non-existent setting indicates embargoes are not supported. A value of -1 allows embargoes of any length. 

Or would  that be considered duplicating documentation that already exists on Dataverse?

Reply:
In the Native API documentation, a link is provided to the MaxEmbargoDurationInMonths page, where the returned values are explained. I think we can adopt the same approach here by adding a link to the Dataverse documentation, where the property is explained